template/python3-http/index.py

```diff
@@ -2,11 +2,20 @@
 from flask import Flask, request, jsonify
 from waitress import serve
 import os
+import sys
+
+import signal
 
 from function import handler
 
 app = Flask(__name__)
 
+def SignalHandler(SignalNumber, Frame):
```

Comment:
Recording my thoughts from the community call here. 

I don't think this is required in any of the templates. The watchdog is capable of implementing the required graceful shutdown logic without any changes or even knowledge in the specific template. 

My definition of graceful shutdown is that 

1. the orchestration sends the signal to stop the service/function.
2. the function should now reject any new requests
3. the function should start a timer and allow any already running requests to finish
4. the function stops when (a) all requests are completed before the timer or (b) the timer is reached and the function forcefully stops

Due to the design of the watchdog, it is capable of implementing all of this logic because it handles all of the requests and the signals. Once we are satisfied with the graceful shutdown implementation in `of-watchdog`, then any template that uses the watchdog should be considered as having a graceful shutdown.